### PR TITLE
Add Slots to atcore

### PR DESCRIPTION
**Pull request by sithlord48**
Add some slots and signals to atcore.
change up the text client to work better with the slots

### CODE REVIEW

File: src/atcore.cpp [hunk 1]
```diff
@@ -172,9 +182,12 @@ void AtCore::print(const QString &fileName)
     while (!gcodestream.atEnd()) {
         QCoreApplication::processEvents(); //make sure all events are processed.
         switch (state()) {
+        case STARTPRINT:
         case IDLE:
         case BUSY:
-            setState(BUSY);
+            if (state() != BUSY) {
```
Comment (patrickelectric) [inline src/atcore.cpp]: Why this ?
The set state function cannot do state != newState, state = newstate ?

File: src/atcore.h [hunk 2]
```diff
@@ -14,7 +14,9 @@ enum PrinterState {
     BUSY, //Printer is working
     PAUSE, //Printer is paused
     ERROR, // Printer Returned Error
-    STOP // Emergency Stop
+    STOP, // Emergency Stop
+    STARTPRINT, //Just Starting a print job
+    FINISHEDPRINT //Just Finished print job
```
Comment (patrickelectric) [inline src/atcore.h]: add comma in the end of this line

**Review by patrickelectric — CHANGES_REQUESTED** [review]

File: src/atcore.cpp (hunk 1 above)
Comment (sithlord48) [reply src/atcore.cpp]: I have removed that line and done as suggested and made the check part of setState

File: src/atcore.cpp [hunk 3]
```diff
@@ -157,9 +157,19 @@ float AtCore::percentagePrinted()
 {
     return percentage;
 }
-
```
Comment (patrickelectric) [inline src/atcore.cpp:160]: Add this line back

File: src/atcore.cpp [hunk 4]
```diff
@@ -172,6 +182,7 @@ void AtCore::print(const QString &fileName)
     while (!gcodestream.atEnd()) {
         QCoreApplication::processEvents(); //make sure all events are processed.
         switch (state()) {
+        case STARTPRINT:
         case IDLE:
         case BUSY:
             setState(BUSY);
```
Comment (patrickelectric) [inline src/atcore.cpp:188]: Remove this line.

**Review by patrickelectric — CHANGES_REQUESTED** [review]

**Pull request merged**